### PR TITLE
docs: add commit guidelines

### PR DESCRIPTION
**Pull request by kenji-miyake**
Related: #6

### CODE REVIEW

File: docs/contributing/pull-request-guidelines/commit-guidelines.md [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+### Follow `Conventional Commits` (required, automated)
+
+#### Rationale
+
+- It can generate categorized changelog, for example using [git-cliff](https://github.com/orhun/git-cliff).
+
+#### Example
+
+```text
+feat(trajectory_follower): add an awesome feature
```
Comment (isamu-takagi) [inline docs/contributing/pull-request-guidelines/commit-guidelines.md:100]: https://www.conventionalcommits.org/en/v1.0.0/#are-the-types-in-the-commit-title-uppercase-or-lowercase
Many [PRs](https://github.com/autowarefoundation/autoware.universe/pulls) start with a lowercase letter and have no period, so let's add that as the recommendation.

Comment (kenji-miyake) [reply docs/contributing/pull-request-guidelines/commit-guidelines.md:100]: Okay, I'll add it later!

Comment (kenji-miyake) [reply docs/contributing/pull-request-guidelines/commit-guidelines.md:100]: How about this? 4f607143e322ec12daa27951e179119cd3a8342c

Comment (isamu-takagi) [reply docs/contributing/pull-request-guidelines/commit-guidelines.md:100]: Looks good.

**Review by isamu-takagi — APPROVED** [review]
LGTM.

**Review by xmfcx — APPROVED** [review]
@kenji-miyake thanks for all the work. Let's merge this and we can iterate over it as we proceed.

**Pull request merged**